### DISCUSSION TITLE
Using A/B Street to show micro examples of transport policy

### DISCUSSION THREAD

**Discussion by hgomersall** [dabreegster/abstreet#691] [General]
We've been discussing using A/B street to showcase small subsets of a transport system to show how specific optimisations might lead to different outcomes. Say, showing how moving 5% of car drivers on to buses eliminates congestion. How easy would it be to build such scenarios with A/B street?

**Comment by dabreegster** [comment]
If you start with a scenario (travel demand model), you can randomly change some percentage of people to use a different mode. https://a-b-street.github.io/docs//user/asu.html#modifying-a-scenario has instructions for doing this. If you can get the baseline simulation to run without gridlock, then you can measure relative changes with this 5% shift scenario -- like which trips get faster or slower, and if people encounter more or less safety issues on their trip.

Which 5% of drivers would switch to buses, though? Probably ones who live near a bus route that you're interested in improving? If this is what you're interested in, luckily I'm soon starting work on [mode shifting](https://github.com/a-b-street/abstreet/issues/448) in response to changes to a map. There's a huge design space for deciding when somebody will decide to change modes, though -- how do you give a score for taking transit vs driving for one trip? How much better do you have to shift that score to convince someone to switch?

If you're specifically interested in modeling people using buses, currently that support isn't ready. @tnederlof is just now looking into importing bus routes from GTFS. We had an earlier implementation using routes from OpenStreetMap, but it's broken.

Any more specifics about what you'd like to try -- what city, what kind of comparison (the 5% shift? or editing the map to convince 5% of people to shift?), and what kind of measurements you'd like to get?

**Reply by hgomersall** [reply]
The thinking is all around Cambridge, UK, but really the intent is to provide policymakers with a few simple examples of how minor changes in policy can lead to large changes in outcome - essentially providing a visualisation of abstract transport decisions. At this stage, the details are not too important, we were hoping to use it as an educational tool rather than as a precise planning tool - "oh look, if I run this scenario, all the little animated cars get stuck, but in this scenario with 5% reduction in car volume, they move easily".

**Reply by dabreegster** [reply]
Got it -- this is the initial pitch. We just need to get a full day running without getting stuck, so we can use all of the comparison dataviz. Since public transit support isn't ready, I'd propose starting with something like "5% of driving trips to biking" instead.

Currently the study area is this:
![Screenshot from 2021-07-05 10-00-24](https://user-images.githubusercontent.com/1664407/124502243-f1a68180-dd77-11eb-810e-89212199b357.png)
Can we shrink it to focus on a more interesting area? There are some roundabouts and complex junctions that  it'd be best to just avoid entirely. I'm also planning to revisit snapping cyclepaths (#330) this quarter, which would be necessary here.

**Reply by hgomersall** [reply]
There are a few Cambridge maps, though has more work been done on that one? When I played with the South Cambridge one previously, I was seeing deadlocks (apologies I never managed to work through that).

**Reply by hgomersall** [reply]
Yeah, just removing the drivers _somehow_ would be enough to start with (and bikes would be just fine).

**Reply by dabreegster** [reply]
I haven't done any work on the Great Kneighton or Cambridge map in ages. I'll take a look at them as I resume work on intersection consolidation and cyclepath snapping. The more you can narrow down a small area that's still interesting to show to people, the easier it'll be to get initial results.

**Reply by hgomersall** [reply]
I think for the moment it doesn't matter. I'll play with that  map. Thanks!

**Reply by hgomersall** [reply]
It's really neat and very interesting. I think I need to decide what it's worth using for, then I'll see what demos can be produced. :)

**Reply by dabreegster** [reply]
I made progress on fixing up wacky intersections, but it still requires some manual effort, so a large area is a non-starter. If you can give a smaller clip of Cambridge, I'd be happy to polish it up.
